### ISSUE TITLE
Cant update MangaDex

### ISSUE THREAD

**Issue by knight109** [riderkick/FMD#1185]
I already update MangaDex modules, but when i want to update it list, i accidentally download from server. Now i cant update list anymore.
Already try to delete the MangaDex data and update it, but still cant

**Comment by Tmp341** [comment]
Delete or backup **fmd.log** file. Then;

``print(StreamToString(http.document))``

Paste this below this line:

``x=TXQuery.Create(http.document)``

inside of Mangadex.lua and try again. If your logging is disabled, enable it before trying.

It should look like this:

> x=TXQuery.Create(http.document)
print(StreamToString(http.document))

Then open your log and paste here.

**Comment by Tmp341** [comment]
I think that too.

**Comment by nomakewan** [comment]
Not the original commenter, but have same issue. Was running the latest MangaDex LUA that supposedly fixed the issue and attempted to update MangaDex for the first time ever only to have nothing show up. Adding the log parameter nets this result:

INFO: <!DOCTYPE html>



 <html class="no-js" lang="en-US"> 
<head>
<title>Access denied | mangadex.org used Cloudflare to restrict access</title>
<meta charset="UTF-8" />
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
<meta http-equiv="X-UA-Compatible" content="IE=Edge,chrome=1" />
<meta name="robots" content="noindex, nofollow" />
<meta name="viewport" content="width=device-width,initial-scale=1,maximum-scale=1" />
<link rel="stylesheet" id="cf_styles-css" href="/cdn-cgi/styles/cf.errors.css" type="text/css" media="screen,projection" />

<style type="text/css">body{margin:0;padding:0}</style>


<script type="text/javascript" src="/cdn-cgi/scripts/zepto.min.js"></script>
<script type="text/javascript" src="/cdn-cgi/scripts/cf.common.js"></script>



</head>
<body>
  <div id="cf-wrapper">
    <div class="cf-alert cf-alert-error cf-cookie-error" id="cookie-alert" data-translate="enable_cookies">Please enable cookies.</div>
    <div id="cf-error-details" class="cf-error-details-wrapper">
      <div class="cf-wrapper cf-header cf-error-overview">
        <h1>
          <span class="cf-error-type" data-translate="error">Error</span>
          <span class="cf-error-code">1006</span>
          <small class="heading-ray-id">Ray ID: 40ba01ad4a659829 &bull; 2018-04-14 23:38:31 UTC</small>
        </h1>
        <h2 class="cf-subheadline">Access denied</h2>
      </div>

      <section></section>

      <div class="cf-section cf-wrapper">
        <div class="cf-columns two">
          <div class="cf-column">
            <h2 data-translate="what_happened">What happened?</h2>
            <p>The owner of this website (mangadex.org) has banned your IP address (IP).</p>
          </div>

          
        </div>
      </div>

      <div class="cf-error-footer cf-wrapper">
  <p>
    <span class="cf-footer-item">Cloudflare Ray ID: <strong>40ba01ad4a659829</strong></span>
    <span class="cf-footer-separator">&bull;</span>
    <span class="cf-footer-item"><span>Your IP</span>: IP</span>
    <span class="cf-footer-separator">&bull;</span>
    <span class="cf-footer-item"><span>Performance &amp; security by</span> <a href="https://www.cloudflare.com/5xx-error-landing?utm_source=error_footer" id="brand_link" target="_blank">Cloudflare</a></span>
    
  </p>
</div>


    </div>
  </div>

  <script type="text/javascript">
  window._cf_translation = {};
  
  
</script>

</body>
</html>


I assume this is the same issue everyone else is having. Sadly once this ban is in place, not even manually adding to favorites works anymore and my IP hasn't changed in forever.

**Comment by Tmp341** [comment]
- Changing IP and being careful.
- Asking for lifting the ban.

These are the options.

**Comment by nomakewan** [comment]
"Be careful" is not a particularly useful suggestion since we're on GitHub. I would personally prefer more precise answers, such as establishing some sort of recommended setting or LUA fix that will poll MangaDex without causing the client to become banned. After all, how is one to 'be careful' when we don't know what 'being careful' actually entails?

**Comment by Tmp341** [comment]
No, I'm talking about connection setting by saying "be careful". Becaus nowadays, most of the sites are known to be using anti spam scripts. And, mangadex probably faces too many requests from both normal users and attackers. There is a limit for a computer too. So, when you download too much at the same time, or "i  think" too fast connection, their script kicks in and bans you. My ISP has "slow" connections, and i use lowest settings. I think that's why i'm still not banned.

**Comment by nomakewan** [comment]
What settings would you suggest, then? I have a fairly quick connection, so I can see why this may have been an issue. It may be very useful to have a recommended setting that's low enough to not earn a ban. Perhaps even make it the program's default setting, and allow the user to manually increase it at their own risk?

**Comment by Tmp341** [comment]
My default settings are:

- ``1`` Number of downloaded tasks at the same time
- ``2`` Number of downloaded files per tasks at the same time

And for mangadex:

- ``1`` MaxTaskLimit
- ``2`` MaxThreadPerTaskLimit
- ``2`` MaxConnectionLimit
- ``1`` UpdateListNumberOfThread
And set a "UserAgent" same as i'm using now.

Not banned since mangadex opened. Haven't tried updating list for a while though.

**Comment by admiralmikan** [comment]
> Set a "UserAgent" same as i'm using now.![2018-04-18 09_34_17-free manga downloader]

Can you please elaborate it more? I don't know what should I write on that section. I got banned today.

(https://user-images.githubusercontent.com/17843605/38908702-b4d77a2c-42eb-11e8-8bb0-6d5b82928918.jpg)

**Comment by kmvi** [comment]
I got a ban even when multithreading is disabled (MaxThreadPerTaskLimit = 1). Сhanging the UserAgent is not enough to circumvent the ban. Also unable to open site in browser. The only method is to change IP. I have no idea how to fix it.

**Comment by Tmp341** [comment]
I don't know why but i can downlod without problem.

But leaving that aside, the only option is communicating and making a deal with mangadex admin.

**Comment by GlassedSilver** [comment]
Didn't get banned but still can't fetch anything.

Used to work until a few days ago. Was on an older version granted, but even going back a few versions, trying again didn't help...

Can't even fetch chapters ("view manga infos") for the entries I do have. My MD favorites all report "0" as current chapter...

welp

**Comment by SBoyer4** [comment]
Same as @GlassedSilver
